scripts/node_bridge.js

```diff
@@ -37,114 +37,115 @@ __BRYTHON__.modules = {}
 jscode = fs.readFileSync(brythonSrcPath, 'utf8');
 eval(jscode);
 
-//function node_import(module,alias,names) {
-function $import_single(module) {
-  var search_path=['../src/libs', '../src/Lib'];
-  var ext=['.js', '.py'];
-  var mods=[module, module+'/__init__'];
-
-  for(var i=0, _len_i = search_path.length; i < _len_i; i++) {
-     for (var j=0, _len_j = ext.length; j < _len_j; j++) {
-         for (var k=0, _len_k = mods.length; k < _len_k; k++) {
-             var path=search_path[i]+'/'+mods[k]+ext[j]
-
-             //console.log("searching for " + path);
-             var module_contents;
-             try {
-               module_contents=fs.readFileSync(path, 'utf8')
-             } catch(err) {}
-             if (module_contents !== undefined) {
-                console.log("imported " + module)
-                //console.log(module_contents);
-                if (ext[j] == '.js') {
-                   return $import_js_module(module,alias,names,path,module_contents)
+function $import_hooks(mod_name, path, from_stdlib) {
+    var search_path = ['./www/src/libs', './www/src/Lib'];
+    var ext = ['.js', '.py'];
+    var mods = [mod_name, mod_name + '/__init__'];
+
+    for (var i = 0, _len_i = search_path.length; i < _len_i; i++) {
+        for (var j = 0, _len_j = ext.length; j < _len_j; j++) {
+            for (var k = 0, _len_k = mods.length; k < _len_k; k++) {
+                var path = search_path[i] + '/' + mods[k] + ext[j]
+                var module_contents;
+                try {
+                    module_contents = fs.readFileSync(path, 'utf8')
+                    if (module_contents !== undefined) {
+                        if (ext[j] === '.js') {
+                            return $import_js_module(mod_name, alias, names, path, module_contents)
+                        }
+                        __BRYTHON__.$py_module_path[mod_name] = path
+                        __BRYTHON__.imported[mod_name] = path
+                        return module_contents
+                    }
+                } catch (err) {
```

Comment:
I think it would be better to only wrap `readFileSync` in a try/catch. As it is now, any error that happens when trying to import a JS module is silently ignored.

Reply:
Yes, but if put logic after `readFileSync` it would be like transaction and we remove one condition checking if file missed, that could improve little bit perfomance